[BUG] ArrayIndexOutOfBounds when reading iceberg table.

**Describe the bug**
Some customer trying to read iceberg table in 25.10, and they see this error:

> 25/11/18 21:20:17 WARN TaskSetManager: Lost task 56.3 in stage 12.0 (TID 210) (172.18.0.4 executor 0): TaskKilled (Stage cancelled: Job aborted due to stage failure: Task 7 in stage 12.0 failed 4 times, most recent failure: Lost task 7.3 in stage 12.0 (TID 181) (172.18.0.4 executor 0): java.util.concurrent.ExecutionException: java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0
> 	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122)
> 	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191)
> 	at com.nvidia.spark.rapids.MultiFileCloudPartitionReaderBase.getNextBuffersAndMetaSingleFile(GpuMultiFileReader.scala:590)
> 	at com.nvidia.spark.rapids.MultiFileCloudPartitionReaderBase.getNextBuffersAndMeta(GpuMultiFileReader.scala:604)
> 	at com.nvidia.spark.rapids.MultiFileCloudPartitionReaderBase.$anonfun$next$1(GpuMultiFileReader.scala:686)
> 	at com.nvidia.spark.rapids.MultiFileCloudPartitionReaderBase.$anonfun$next$1$adapted(GpuMultiFileReader.scala:635)
> 	at com.nvidia.spark.rapids.Arm$.withResource(Arm.scala:30)
> 	at com.nvidia.spark.rapids.MultiFileCloudPartitionReaderBase.next(GpuMultiFileReader.scala:635)
> 	at com.nvidia.spark.rapids.iceberg.parquet.SingleFileColumnarBatchIterator.hasNext(GpuMultiThreadIcebergParquetReader.scala:153)
> 	at com.nvidia.spark.rapids.iceberg.parquet.GpuMultiThreadIcebergParquetReader.hasNext(GpuMultiThreadIcebergParquetReader.scala:65)
> 	at org.apache.iceberg.spark.source.GpuIcebergPartitionReader.next(GpuIcebergPartitionReader.scala:68)
> 	at com.nvidia.spark.rapids.PartitionIterator.hasNext(dataSourceUtil.scala:29)
> 	at com.nvidia.spark.rapids.MetricsBatchIterator.hasNext(dataSourceUtil.scala:46)
> 	at com.nvidia.spark.rapids.shims.GpuDataSourceRDD$$anon$1.$anonfun$hasNext$1(GpuDataSourceRDD.scala:81)
> 	at com.nvidia.spark.rapids.shims.GpuDataSourceRDD$$anon$1.$anonfun$hasNext$1$adapted(GpuDataSourceRDD.scala:81)
> 	at scala.Option.exists(Option.scala:376)
> 	at com.nvidia.spark.rapids.shims.GpuDataSourceRDD$$anon$1.hasNext(GpuDataSourceRDD.scala:81)
> 	at com.nvidia.spark.rapids.shims.GpuDataSourceRDD$$anon$1.advanceToNextIter(GpuDataSourceRDD.scala:105)
> 	at com.nvidia.spark.rapids.shims.GpuDataSourceRDD$$anon$1.hasNext(GpuDataSourceRDD.scala:81)
> 	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:37)
> 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
> 	at com.nvidia.spark.rapids.GpuOpTimeTrackingRDD$$anon$1.$anonfun$hasNext$2(GpuExec.scala:72)
> 	at scala.runtime.java8.JFunction0$mcZ$sp.apply(JFunction0$mcZ$sp.java:23)
> 	at com.nvidia.spark.rapids.GpuMetric.ns(GpuMetrics.scala:354)
> 	at com.nvidia.spark.rapids.GpuMetric.ns(GpuMetrics.scala:347)
> 	at com.nvidia.spark.rapids.GpuOpTimeTrackingRDD$$anon$1.hasNext(GpuExec.scala:72)
> 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:491)
> 	at com.nvidia.spark.rapids.GpuOpTimeTrackingRDD$$anon$1.$anonfun$hasNext$1(GpuExec.scala:68)
> 	at scala.runtime.java8.JFunction0$mcZ$sp.apply(JFunction0$mcZ$sp.java:23)
> 	at com.nvidia.spark.rapids.GpuMetric.ns(GpuMetrics.scala:354)
> 	at com.nvidia.spark.rapids.GpuOpTimeTrackingRDD$$anon$1.hasNext(GpuExec.scala:68)
> 	at com.nvidia.spark.rapids.CollectTimeIterator.$anonfun$hasNext$1(GpuMetrics.scala:405)
> 	at com.nvidia.spark.rapids.CollectTimeIterator.$anonfun$hasNext$1$adapted(GpuMetrics.scala:404)
> 	at com.nvidia.spark.rapids.Arm$.withResource(Arm.scala:30)
> 	at com.nvidia.spark.rapids.CollectTimeIterator.hasNext(GpuMetrics.scala:404)
> 	at com.nvidia.spark.rapids.AbstractGpuCoalesceIterator.getHasOnDeck(GpuCoalesceBatches.scala:315)
> 	at com.nvidia.spark.rapids.AbstractGpuCoalesceIterator.hasNext(GpuCoalesceBatches.scala:332)
> 	at com.nvidia.spark.rapids.GpuOpTimeTrackingRDD$$anon$1.$anonfun$hasNext$1(GpuExec.scala:68)
> 	at scala.runtime.java8.JFunction0$mcZ$sp.apply(JFunction0$mcZ$sp.java:23)
> 	at com.nvidia.spark.rapids.GpuMetric.ns(GpuMetrics.scala:354)
> 	at com.nvidia.spark.rapids.GpuOpTimeTrackingRDD$$anon$1.hasNext(GpuExec.scala:68)
> 	at com.nvidia.spark.rapids.AbstractProjectSplitIterator.hasNext(basicPhysicalOperators.scala:256)
> 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
> 	at com.nvidia.spark.rapids.GpuOpTimeTrackingRDD$$anon$1.$anonfun$hasNext$1(GpuExec.scala:68)
> 	at scala.runtime.java8.JFunction0$mcZ$sp.apply(JFunction0$mcZ$sp.java:23)
> 	at com.nvidia.spark.rapids.GpuMetric.ns(GpuMetrics.scala:354)
> 	at com.nvidia.spark.rapids.GpuOpTimeTrackingRDD$$anon$1.hasNext(GpuExec.scala:68)
> 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
> 	at com.nvidia.spark.rapids.GpuOpTimeTrackingRDD$$anon$1.$anonfun$hasNext$1(GpuExec.scala:68)
> 	at scala.runtime.java8.JFunction0$mcZ$sp.apply(JFunction0$mcZ$sp.java:23)
> 	at com.nvidia.spark.rapids.GpuMetric.ns(GpuMetrics.scala:354)
> 	at com.nvidia.spark.rapids.GpuOpTimeTrackingRDD$$anon$1.hasNext(GpuExec.scala:68)
> 	at com.nvidia.spark.rapids.CollectTimeIterator.$anonfun$hasNext$1(GpuMetrics.scala:405)
> 	at com.nvidia.spark.rapids.CollectTimeIterator.$anonfun$hasNext$1$adapted(GpuMetrics.scala:404)
> 	at com.nvidia.spark.rapids.Arm$.withResource(Arm.scala:30)
> 	at com.nvidia.spark.rapids.CollectTimeIterator.hasNext(GpuMetrics.scala:404)
> 	at com.nvidia.spark.rapids.CloseableBufferedIterator.hasNext(CloseableBufferedIterator.scala:68)
> 	at org.apache.spark.sql.rapids.execution.GpuBroadcastHelper$.$anonfun$getBroadcastBuiltBatchAndStreamIter$2(GpuBroadcastHelper.scala:125)
> 	at com.nvidia.spark.rapids.Arm$.withResource(Arm.scala:30)
> 	at org.apache.spark.sql.rapids.execution.GpuBroadcastHelper$.$anonfun$getBroadcastBuiltBatchAndStreamIter$1(GpuBroadcastHelper.scala:124)
> 	at com.nvidia.spark.rapids.Arm$.closeOnExcept(Arm.scala:98)
> 	at org.apache.spark.sql.rapids.execution.GpuBroadcastHelper$.getBroadcastBuiltBatchAndStreamIter(GpuBroadcastHelper.scala:119)
> 	at org.apache.spark.sql.rapids.execution.GpuBroadcastHashJoinExecBase.$anonfun$doColumnarBroadcastJoin$1(GpuBroadcastHashJoinExecBase.scala:151)
> 	at org.apache.spark.rdd.RDD.$anonfun$mapPartitions$2(RDD.scala:858)
> 	at org.apache.spark.rdd.RDD.$anonfun$mapPartitions$2$adapted(RDD.scala:858)
> 	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
> 	at com.nvidia.spark.rapids.GpuOpTimeTrackingRDD.compute(GpuExec.scala:61)
> 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
> 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
> 	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
> 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
> 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
> 	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
> 	at com.nvidia.spark.rapids.GpuOpTimeTrackingRDD.compute(GpuExec.scala:61)
> 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
> 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
> 	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
> 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
> 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
> 	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
> 	at com.nvidia.spark.rapids.GpuOpTimeTrackingRDD.compute(GpuExec.scala:61)
> 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
> 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
> 	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
> 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
> 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
> 	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
> 	at com.nvidia.spark.rapids.GpuOpTimeTrackingRDD.compute(GpuExec.scala:61)
> 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
> 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
> 	at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
> 	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:104)
> 	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:54)
> 	at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:166)
> 	at org.apache.spark.scheduler.Task.run(Task.scala:141)
> 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$4(Executor.scala:621)
> 	at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally(SparkErrorUtils.scala:64)
> 	at org.apache.spark.util.SparkErrorUtils.tryWithSafeFinally$(SparkErrorUtils.scala:61)
> 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:94)
> 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:624)
> 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
> 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
> 	at java.base/java.lang.Thread.run(Thread.java:840)
> Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0
> 	at com.nvidia.spark.rapids.iceberg.parquet.GpuIcebergParquetReader.$anonfun$filterParquetBlocks$1(reader.scala:199)
> 	at com.nvidia.spark.rapids.Arm$.withResource(Arm.scala:30)
> 	at com.nvidia.spark.rapids.iceberg.parquet.GpuIcebergParquetReader.filterParquetBlocks(reader.scala:195)
> 	at com.nvidia.spark.rapids.iceberg.parquet.GpuIcebergParquetReader.filterParquetBlocks$(reader.scala:193)
> 	at com.nvidia.spark.rapids.iceberg.parquet.GpuMultiThreadIcebergParquetReader.filterBlock(GpuMultiThreadIcebergParquetReader.scala:131)
> 	at com.nvidia.spark.rapids.iceberg.parquet.GpuMultiThreadIcebergParquetReader.$anonfun$createParquetReader$2(GpuMultiThreadIcebergParquetReader.scala:96)
> 	at com.nvidia.spark.rapids.parquet.MultiFileCloudParquetPartitionReader$ReadBatchRunner.doRead(GpuParquetScan.scala:2728)
> 	at com.nvidia.spark.rapids.parquet.MultiFileCloudParquetPartitionReader$ReadBatchRunner.call(GpuParquetScan.scala:2700)
> 	at com.nvidia.spark.rapids.parquet.MultiFileCloudParquetPartitionReader$ReadBatchRunner.call(GpuParquetScan.scala:2680)
> 	at com.nvidia.spark.rapids.MultiFileCloudPartitionReaderBase$$anon$1.call(GpuMultiFileReader.scala:402)
> 	at com.nvidia.spark.rapids.MultiFileCloudPartitionReaderBase$$anon$1.call(GpuMultiFileReader.scala:394)
> 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
> 	... 3 more



**Steps/Code to reproduce bug**


**Expected behavior**

Should work.

--- a/iceberg/src/main/scala/com/nvidia/spark/rapids/iceberg/parquet/reader.scala
+++ b/iceberg/src/main/scala/com/nvidia/spark/rapids/iceberg/parquet/reader.scala
@@ -199,9 +199,8 @@ trait GpuIcebergParquetReader extends Iterator[ColumnarBatch] with AutoCloseable
       val fileSchema = reader.getFileMetaData.getSchema
 
       val rowGroupFirstRowIndices = new Array[Long](reader.getRowGroups.size())
-      rowGroupFirstRowIndices(0) = 0
-      var accumulatedRowCount = reader.getRowGroups.get(0).getRowCount
-      for (i <- 1 until reader.getRowGroups.size()) {
+      var accumulatedRowCount = 0L
+      for (i <- 0 until reader.getRowGroups.size()) {
         rowGroupFirstRowIndices(i) = accumulatedRowCount
         accumulatedRowCount += reader.getRowGroups.get(i).getRowCount
       }

